### DISCUSSION TITLE
appmenu support for global menus

### DISCUSSION THREAD

**Discussion by probonopd** [CuarzoSoftware/Louvre#15] [Ideas]
It would be nice if application menus could be rendered in the global menu bar at the top of the screen.

https://wayland.app/protocols/kde-appmenu (even though it says "kde" in its name, it should not be limited to KDE applications)

**Comment by ehopperdietzel** [comment]
So, does this allow apps to use the dbusmenu like in X11? Does it enable apps to obtain a "window id"?

**Reply by probonopd** [reply]
Looks like KDE is using it to get global menus working on Wayland (it is working there).
The protocol is defined at https://invent.kde.org/libraries/plasma-wayland-protocols/-/blob/master/src/protocols/appmenu.xml.

**Reply by ehopperdietzel** [reply]
I will try KDE to see how well it works.

**Reply by probonopd** [reply]
Also check https://github.com/Alexays/Waybar/issues/288 - maybe you can work together, if only for documenting how all of this is supposed to work under Wayland. (I'd happily add a section about Wayland to my [collection](https://hellosystem.github.io/docs/developer/menu.html) of global menu related "documentation" or lack thereof.)

**Reply by ehopperdietzel** [reply]
Great, I'll begin creating my own Wayland compositor with Louvre shortly after adding some features. Afterward, I'll investigate how to implement a global menu using this. Once I understand it, I'll be happy to help document it and provide implementation examples.

**Reply by ehopperdietzel** [reply]
Also, I want to change its license to MIT in case you want to use it, but I first need to get rid of freeimage.

**Reply by ehopperdietzel** [reply]
I am also curious. There is a lot of functionality in the SRM graphic backend that relies on the Linux DRM/KMS API. Is that available in FreeBSD as well?

**Reply by probonopd** [reply]
https://wiki.freebsd.org/Graphics#DRM has the details.

**Reply by ehopperdietzel** [reply]
Nice, so it would work, but it seems the DRM prime API is not yet ported (for better multi-GPU performance). Not a big deal I guess.
